src/execution.jl

```diff
@@ -253,6 +253,13 @@ end
             if !ondisk_hit && path !== nothing && disk_cache_enabled()
                 @debug "Writing out on-disk cache" job path
                 mkpath(dirname(path))
+                if haskey(asm[2], :gv_to_value)
+                    # TODO: Serializer cannot handle Core.IntrinsicFunction
+                    # We kinda want Julia to serialize the values in `gv_to_value` in the pkgimg and us just having to store an index
+                    # for now we just empty them out
+                    # We would need to remove the initializers from LLVM IR as well to be correct, and then link these in at runtime
+                    empty!(asm[2].gv_to_value)
```

Comment:
We should likely just give up on caching this for now if `!isempty(gv_to_value)`

For proper caching, we would want to create an `IdDict` in the current parent module  and instead of saving values we would want to save "offsets/keys" into that `IdDict` so that Julia performs the pkgimg reolcation for us.